packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts

```diff
@@ -209,13 +232,13 @@ export class DbSyncStakePoolProvider extends DbSyncProvider(RunnableModule) impl
         this.getPoolsDataOrdered(poolUpdates, totalAdaAmount, options)
       );
     // Get stake pools APYs cached
-    const poolAPYs =
-      sortType === 'apy'
+    const poolAPYs = this.response?.search?.metrics?.apy
+      ? sortType === 'apy'
         ? (orderedResult as PoolAPY[])
         : await this.#cache.get(queryCacheKey(StakePoolsSubQuery.APY, hashesIds, options), () =>
             this.#builder.queryPoolAPY(hashesIds, this.#epochLength, { rewardsHistoryLimit })
-          );
-
+          )
+      : [];
```

Comment:
Might need a new status code though
```suggestion
      : (() => { throw new ProviderError(...); })()
```

Reply:
An error is thrown [here](https://github.com/input-output-hk/cardano-js-sdk/pull/609/files#diff-3a938d2c168d4b251612c6efabf201ac70e7585c3c1c2149b3c34e71981e6666R112-R114) if the request specifies to sort by apy when the response is not configured to return it. It's covered by tests